src/Market/Onchain.hs

```diff
@@ -77,18 +78,28 @@ mkBuyValidator pkh nfts r ctx =
     sig :: PubKeyHash
     sig = case txInfoSignatories info of
             [pubKeyHash] -> pubKeyHash
+            _ -> error ()
 
     price :: Integer
     price = nPrice nfts
 
     checkNFTOut :: Bool
     checkNFTOut = valueOf (valuePaidTo info sig) cs tn == 1
-    
+
+    marketplaceFees :: Integer
+    marketplaceFees = 20
+
+    sellerPercentage :: Integer
+    sellerPercentage = (1000 - marketplaceFees) - nRoyaltyPercent nfts
+
     checkSellerOut :: Bool
-    checkSellerOut = fromInteger (Ada.getLovelace (Ada.fromValue (valuePaidTo info seller))) >= 98 % 100 * fromInteger price
+    checkSellerOut = fromInteger (Ada.getLovelace (Ada.fromValue (valuePaidTo info seller))) >= sellerPercentage % 1000 * fromInteger price
```

Comment:
We need to `checkSellerOut` as the last step by subtracting the fees & royalties from the previous steps. There's a bug right now for assets sold for less than 50 ADA.

Ex. Selling an asset for 10 ADA
max(1 ADA, marketplaceFee) will require 1 ADA from the purchase price
max(1 ADA, royaltyFee) will require 1 ADA from the purchase price, even if royalty is 0% (this is a second bug)
sellerReceives 8 ADA, but the script is expecting them to receive 10 * 0.98 = 9.80 ADA.